Src/IronPython/Runtime/Operations/PythonOps.cs

```diff
@@ -1390,36 +1358,82 @@ public static partial class PythonOps {
                 if (bases.Count == 0) {
                     bases = PythonTuple.MakeTuple(DynamicHelpers.GetPythonTypeFromType(typeof(object)));
                 }
-                return PythonType.__new__(context, TypeCache.PythonType, name, bases, vars, selfNames);
+                PythonDictionary vars = func(parentContext).Dict;
+                return PythonType.__new__(parentContext, TypeCache.PythonType, name, bases, vars, selfNames);
             }
 
-            object? classdict = vars;
-
-            if (metaclass is PythonType metatype) {
-                classdict = CallPrepare(context, metatype, name, bases, keywords, vars);
-            }
+            // Prepare classsdict
+            // TODO: prepared classdict should be used by `func` (PEP 3115)
+            object? classdict = callPrepare(parentContext, metaclass, name, bases, keywords, func(parentContext).Dict);
 
-            // eg:
-            // def foo(*args): print(args)
-            // class bar(metaclass=foo): pass
+            // Dispatch to the metaclass to do class creation and initialization
+            // metaclass could be simply a callable, eg:
+            // >>> def foo(*args): print(args)
+            // >>> class bar(metaclass=foo): pass
             // calls our function...
-            PythonContext pc = context.LanguageContext;
+            PythonContext pc = parentContext.LanguageContext;
 
             object obj = pc.MetaClassCallSite.Target(
                 pc.MetaClassCallSite,
-                context,
+                parentContext,
                 metaclass,
                 name,
                 bases,
                 classdict,
-                keywords
+                keywords ?? MakeEmptyDict()
             );
 
+            // Ensure the class derives from `object`
             if (obj is PythonType newType && newType.BaseTypes.Count == 0) {
                 newType.BaseTypes.Add(DynamicHelpers.GetPythonTypeFromType(typeof(object)));
             }
 
             return obj;
+            // ------------------------------------------------------------------------
+
+            static Func<CodeContext, CodeContext> getClassCode(CodeContext/*!*/ context, FunctionCode funcCode, Func<CodeContext, CodeContext> body) {
```

Comment:
Not that you need to change this, but I usually PascalCase local functions. We should probably adopt a convention. I'll have to check if there's an editorconfig rule for this.

Reply:
I usually use dromedaryCase for anything local (inc. functions) to distinguish it from regular methods. But I am happy to adopt any convention; .editorconfig check would be good. However, I may have committed my dromedaryCase in some other places already, though now I can't find it...

Reply:
I managed to define an .editorconfig rule, unfortunately it seems like the file has to be open in VS for it to display an issue. Anyway, I opened a bunch of files to see if it would find rule violation and there are indeed a few. An exception to the rule might be the helper methods like `link.linkWindows` where the casing matches the public parent. Another interesting case is where the local function is essentially replacing lambda (e.g. in `SurrogatePassErrors`). I assume I would have dromedaryCased the lambda? I guess it's not as clear cut as I thought it was.